example/mailto/mailto.cpp

```diff
@@ -0,0 +1,377 @@
+    Unlike a urls::url_view, which only represents
+    the general syntax of urls, a mailto_view
+    represents a reference to fields that are
+    relevant to mailto URLs, while ignoring
+    elements of the general syntax
+    that are not relevant to the scheme.
+
+    This allows us to use the general syntax
+    parsers to create a representation that
+    is more appropriate for the specified scheme
+    syntax.
+
+    @par Specification
+    @li <a href="https://www.rfc-editor.org/rfc/rfc6068"
+        >The 'mailto' URI Scheme</a>
+    @li <a href="https://www.rfc-editor.org/rfc/rfc2368"
+        >The mailto URL scheme</a>
+    @li <a href="https://www.rfc-editor.org/rfc/rfc1738"
+        >Uniform Resource Locators (URL)</a>
+
+    @par References
+    @li <a href="https://en.wikipedia.org/wiki/Mailto"
+        >mailto (Wikipedia)</a>
+
+ */
+class mailto_view
+{
+    urls::url_view u_;
+
+public:
+    /// Return the specified email address in the URL
+    /**
+        A mailto URL might contain multiple email
+        addresses separated by commas.
+
+        The first addresses are represented in
+        the path. Other addresses are in
+        any query parameter whose key is "to".
+
+        @param idx Address index
+
+        @return The specified address
+     */
+    std::string
+    address(std::size_t idx = 0) const
+    {
+        return encoded_address(idx).decode();
+    }
+
+    /// @copydoc address()
+    urls::pct_string_view
+    encoded_address(std::size_t idx = 0) const noexcept
+    {
+        // Look for ith email address in the path string
+        auto s = ith_address(u_.encoded_path(), idx);
+        if (s)
+            return *s;
+
+        // Look for ith email address in one of the "to" headers
+        auto ps = u_.encoded_params();
+        auto it = ps.find("to", urls::ignore_case);
+        while (it != ps.end())
+        {
+            s = ith_address((*it++).value, idx);
+            if (s)
+                return *s;
+            it = ps.find(it, "to", urls::ignore_case);
+        }
+        return {};
+    }
+
+    /// Return number of email addresses in the URL
+    std::size_t
+    size() const noexcept
+    {
+        // Count addresses in path
+        std::size_t n = addr_in_str(u_.encoded_path());
+
+        // Count addresses in "to" headers
+        auto ps = u_.encoded_params();
+        auto it = ps.find("to", urls::ignore_case);
+        while (it != ps.end())
+        {
+            n += addr_in_str((*it++).value);
+            it = ps.find(it, "to", urls::ignore_case);
+        }
+        return n;
+    }
+
+    /// Return the specified cc email address in the URL
+    /**
+        A mailto URL might contain multiple cc
+        email addresses separated by commas.
+
+        Addresses can be represented in any query
+        parameter whose key is "cc".
+
+        @param idx Address index
+
+        @return The specified cc address
+     */
+    std::string
+    cc(std::size_t idx) const
```

Comment:
Can we just say `i` or `index` and not use the abbreviation which is unpronounceable?